app/src/interfaces/input-code/input-code.vue

```diff
@@ -119,6 +119,10 @@ export default defineComponent({
 				return JSON.stringify(props.value, null, 4);
 			}
 
+			if (typeof props.value === 'number' || typeof props.value === 'boolean') {
+				return JSON.stringify(props.value);
+			}
```

Comment:
Why you want to convert boolean or number into strings?
They are valid types on JSON

Reply:
Because that's what the function does, it's to pass it into code-mirror. which only accepts strings.

Reply:
The code above is inside of `		const stringValue = computed<string>(() => {
			if (props.value === null) return '';

			if (typeof props.value === 'object') {
				return JSON.stringify(props.value, null, 4);
			}

			if (typeof props.value === 'number' || typeof props.value === 'number') {
				return JSON.stringify(props.value);
			}

			return props.value as string;
		});`

Reply:
@joselcvarela @rijkvanzanten Might of been confused by my title, i forgot to include "Not"

Reply:
Thank you,
Can you provide a short video of what happened before and what happens with this change?
This helps a lot to move on with PRs 👍

Reply:
If you go to 20 seconds in the video https://streamable.com/34tbac you'll see that code-mirror doesn't render the value, as it's not passed into it, as a string.  the code I've pushed, fixes that. and it gets rendered properly https://i.imgur.com/tH9Scdm.png @joselcvarela

Reply:
@joselcvarela The issue you're having with strings, is also because most databases don't allow strings to be put in directly. they support objects, arrays, numbers, booleans, but strings must be wrapped in quotation marks. see https://streamable.com/vr8pka

Reply:
Thank you so much @cory-glooh it really helps a lot 👍
About the strings, we should handle them before they hit the database in order to properly save them

Reply:
With a bit of code you can achieve a working result, but it's not ideal and not how I suspect you would want it done. https://streamable.com/vad2m4

Reply:
Why you say it's not ideal? String is a valid JSON type 🤔

Reply:
String is a valid json type, but only when encased in quotes. when parsing a string, or inserting  a string into a json field in a db, it needs to be encased.

Reply:
JSON.parse("hello") fails, as does inserting "hello" into a database. but '"hello"' works for both.

Reply:
Can you handle strings as well?
This should be an app issue only.
I believe we should handle all valid JSON types in the app side. Then API handles the rest.
If you do so, we could close https://github.com/directus/directus/issues/11098 😊

Reply:
@cory-glooh Any thoughts on the above? 🙂